common/logger/src/struct_log.rs

```diff
@@ -476,8 +481,17 @@ impl TCPStructLogThread {
         }
     }
 
-    pub fn connect(&self) -> io::Result<TcpStream> {
-        TcpStream::connect_timeout(&self.address, Duration::from_millis(CONNECTION_TIMEOUT_MS))
+    pub fn connect(&mut self) -> io::Result<TcpStream> {
+        // resolve addresses to handle DNS names
+        let addrs: Vec<SocketAddr> = self.address.to_socket_addrs()?.collect();
+        if let Some(addr) = addrs.choose(&mut self.rand) {
+            TcpStream::connect_timeout(addr, Duration::from_millis(CONNECTION_TIMEOUT_MS))
```

Comment:
I would recommend just iterating through the list, in general, my experience has been that this list will contain an ipv6 and an ipv4 address and you're going to get a connection refused if the app only binds to one or the either.